### PR TITLE
ref: await linkprofile before adding data, throw if not signed

### PR DESCRIPTION
**Pull request by zachferland**

### CODE REVIEW

File: src/3box.js [hunk 1]
```diff
@@ -488,8 +488,12 @@ class Box {
     if (!linkData) {
       const address = this._3id.managementAddress
       const did = this._3id.muportDID
-
-      const consent = await utils.getLinkConsent(address, did, this._web3provider)
+      let consent
+      try {
+        consent = await utils.getLinkConsent(address, did, this._web3provider)
+      } catch (e) {
+        throw new Error('Link consent message must be signed before adding data, to link address to store')
```
Comment (oed) [inline src/3box.js]: Nice addition!

File: src/3box.js [hunk 2]
```diff
@@ -506,11 +510,7 @@ class Box {
     }
 
     // Send consentSignature to 3box-address-server to link profile with ethereum address
-    try {
-      await utils.fetchJson(this._serverUrl + '/link', linkData)
-    } catch (err) {
-      console.error(err)
-    }
+    utils.fetchJson(this._serverUrl + '/link', linkData).catch(console.log)
```
Comment (oed) [inline src/3box.js]: Why console.log and not console.error? Or maybe warn?

Comment (zachferland) [reply src/3box.js]: yeah just changed that

Comment (oed) [reply src/3box.js]: haha, you fixed it before I finished the review :P

Comment (zachferland) [reply src/3box.js]: we are too quick now!

tests still not happy about it yet

**Pull request merged**